packages/contracts/src/dollar/core/StakingShare.sol

```diff
@@ -1,13 +1,19 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.16;
 
-import {ERC1155Ubiquity} from "./ERC1155Ubiquity.sol";
 import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
-import "../../../src/dollar/utils/SafeAddArray.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155URIStorage.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Burnable.sol";
+import "@openzeppelin/contracts/token/ERC1155/extensions/ERC1155Pausable.sol";
+import "../../dollar/utils/SafeAddArray.sol";
 import "../interfaces/IAccessControl.sol";
 import "../libraries/Constants.sol";
 
-contract StakingShare is ERC1155Ubiquity {
+contract StakingShare is
```

Comment:
Here you should inherit from `ERC1155Ubiquity , ERC1155URIStorage `

The `ERC1155Ubiquity` contract is `ERC1155Pausable` and `ERC1155Burnable` under the hood. It also has access control. So if you inherit from `ERC1155Ubiquity` then you can delete:
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L38
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L72
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L131
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L141
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L141
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L162
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L197
- https://github.com/ubiquity/ubiquity-dollar/blob/2926ae730aad94c6090bd016ccc9927537f39a9b/packages/contracts/src/dollar/core/StakingShare.sol#L203

Reply:
Thanks for the review. Some functions are requesting to be overwritten so I let them there for your review. ERC1155URIStorage also overwrites those same functions.